rust/src/conflict_checker.rs

```diff
@@ -0,0 +1,459 @@
+//! Helper module to check if a transaction can be committed in case of conflicting commits.
+use crate::action::{Action, Add, DeltaOperation, MetaData, Protocol, Remove};
+use crate::{
+    table_state::DeltaTableState, DeltaDataTypeTimestamp, DeltaDataTypeVersion, DeltaTable,
+    DeltaTableError, DeltaTableMetaData,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::{Map, Value};
+use std::collections::{HashMap, HashSet};
+use std::io::{BufRead, BufReader, Cursor};
+
+/// The commitInfo is a fairly flexible action within the delta specification, where arbitrary data can be stored.
+/// However the reference implementation as well as delta-rs store useful information that may for instance
```

Comment:
Please add a link to the official documentation on the Delta Lake possible conflicts.

I did find this one — https://docs.databricks.com/optimizations/isolation-level.html#write-conflicts-on-databricks — but I'm not sure how complete it is.